keps/sig-scheduling/3902-decoupled-taint-manager/README.md

```diff
@@ -60,35 +57,38 @@ Items marked with (R) are required *prior to targeting to a milestone / release*
-* There are increasing needs to customize and extend the default node taint-based pod evictions in practice.
-  * Whether or not evict a stateful pod with local storage on `NoExecute` can depend on the actual taint conditions, workload status and other conditions. The decision may require information from the external systems and controllers, which have more knowledge and context about the workloads.  A custom `taint-manager`  will be able to control how and when a pod is evicted or deleted, including making calls to external systems and take actions on  `NoExecute` taints with other conditions such as unhealthy network. 
-  * Starting kubernetes 1.27, with the removal of the flag `enable-taint-manager`, replacing the default `NoExecute` taint-manager with a custom one would be more challenging. `NoExecute` taints can disable the no-execute taint manager, but such disabling can only be achieved through a mutating webhook on all pods, and it can interact negatively with the upstream `DefaultTolerations` webhook.
-  * The requirement of custom taint-based eviction is not unique. It exists already in Kubernetes, where we call it `FullyDistributionMode`. When all nodes are unhealthy, `NodeLifecycleController` chooses not to apply `NoExecute` taints properly. In a real-world case, the integrator may choose to define the disruption rate/budget, etc.
+`NodeLifecycleController` combines two independent functions: 
+   * Adding a pre-defined set of `NoExecute` taints to nodes based on the node conditions 
+   * Performing pod eviction on `NoExecute` taints
+
+Splitting the `NodeLifecycleController` based on the above functionalities will help to disentagle code and make future extensions to either component manageable.
 
 ### Goals
 
-* Move taint-based eviction implementation out of `NodeLifecycleController` into a separate and independent taint-manager to support flexible extension and customization of node taint-based eviction.
+* Move taint-based eviction implementation out of `NodeLifecycleController` into a separate and independent taint-manager to enhance separation of concerns, maintainability. This separation also enables cluster-admins to build custom `TaintManager`s and use them in their clusters.
 
 ### Non-Goals
 
-* Improve and change the APIs to support extension and customization of the existing `TainaManager`.
+* Improve and change the APIs to support extension and customization of the existing `TaintManager`.
 * Actual extension and improvement of the current `TaintManager`.
 
 ## Proposal
 
 ### User Stories (Optional)
 
-#### Story 1
+#### Story
+
+While this split is more of a kubernetes developer focused change, the side-effect of this change is that it allows cluster-admins to develop custom `TaintManager`s and disable the default `TaintManager`. 
```

Comment:
side-effect sounds more like a negative effect, maybe reword to "by-product" or "unintended positive effect".